docs/cudf/source/internals.md

```diff
@@ -1,11 +1,111 @@
 cuDF internals
 ==============
 
-ColumnAccessor
---------------
+The cuDF API closely matches that of the [Pandas](https://pandas.pydata.org/) library.
+Thus, we have the types `cudf.Series`, `cudf.DataFrame` and `cudf.Index` which look and
+feel very much like their Pandas counterparts.
+
+Under the hood, however, cuDF uses data structures very different from Pandas. In this document,
+we describe these internal data structures.
+
+## Column
+
+Columns are cuDF's core data structure and they are modeled after
+the [Apache Arrow Columnar Format](https://arrow.apache.org/docs/format/Columnar.html).
+
+A column represents a sequence of values, any number of which may be "null". Columns are
+specialized based on the type of data they contain. Thus we have `NumericalColumn`, `StringColumn`,
+`DatetimeColumn`, etc.,
+
+A column is composed of the following:
+
+* A **data type**, specifying the type of each element.
+* A **mask buffer** that may store the data for the column elements.
+  Some column types do not have a data buffer, instead storing data in the children columns.
+* A **mask buffer** whose bits represent the validity (null or not null) of each element.
+  Columns whose elements are all "valid" may not have a mask buffer.
+* A tuple of **children** columns, which enable the representation complex types such as
+  columns with non-fixed width elements (e.g., strings, lists).
+* If the Column represents a slice, an integer **offset** into the first element of the slice
```

Comment:
I think we should just explicitly call out that Columns have an `offset` and a `size` so the underlying buffer isn't necessarily 100% representative of the Column itself.

Reply:
Fixed.